src/dashboard-refactor/header/index.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { PureComponent } from 'react'
+import styled from 'styled-components'
+
+import { OVERVIEW_URL } from 'src/constants'
+import SearchBar, { SearchBarProps } from './search-bar'
+import { SyncStatusIconState } from './types'
+import SyncStatusMenu, { SyncStatusMenuProps } from './sync-status-menu'
+import SidebarHeader, { SidebarHeaderProps } from './sidebar-header'
+import styles, { fonts } from '../styles'
+import { Icon } from '../styled-components'
+import Margin from '../components/Margin'
+
+const Container = styled.div`
+    height: 45px;
+    width: 100%;
+    position: fixed;
+    top: 0px;
+    display: flex;
+    flex-direction: row;
+    align-items: center;
+    justify-content: space-between;
+`
+
+const RightHeader = styled.div`
+    width: min-content;
+    display: flex;
+    align-items: center;
+    justify-content: start;
+`
+
+const SyncStatusIcon = styled.div<{
+    color: SyncStatusIconState
+}>`
+    height: 12px;
+    width: 12px;
+    border-radius: 6px;
+    background-color: ${(props) =>
+        styles.components.syncStatusIcon.colors[props.color]};
+`
+
+const SyncStatusHeaderText = styled.span<{
+    textCentered: boolean
+}>`
+    font-family: ${fonts.primary.name};
+    font-weight: ${fonts.primary.weight.bold};
+    color: ${fonts.primary.colors.primary};
+    font-size: 10px;
+    line-height: 15px;
+    white-space: nowrap;
+    overflow: hidden;
+    ${(props) => props.textCentered && 'text-align: center;'}
```

Comment:
This line _may_ be an issue if `textCentered` is falsy. As `props.textCentered && 'text-align: center;'` will evaluate to `false` rather than an empty string or anything else. And I don't think `false` is valid CSS.

We often use `&&` operator in React JSX rendering logic as if it does evaluate to `false` that's still perfectly valid JSX from React's POV (`false` just means "render nothing" to the React renderer IIRC). But it can definitely cause some issues when used in other contexts, maybe here in styled components

I think a safer construct would be `props.textCentered ? 'text-align: center;' : ''`

Reply:
Cool cool, will do.